### DISCUSSION TITLE
How to set the time/timezone?

### DISCUSSION THREAD

**Discussion by jtenniswood** [xblax/flashforge_ad5m_klipper_mod#137] [General]
The time is incorrect on klipperscreen, but I can't figure out how to change it. Could someone point me in the right direction?
Thanks!

**Comment by DominicWeston** [comment]
I haven't tried this yet, but......

cat /root/printer_data/config/KlipperScreen.conf
[main]
show_cursor: False
use_default_menu: True
theme: material-dark
screen_blanking: 3600
screen_brightness: 70
timezone: UTC
font_size: large

[printer FlashForge Adventurer 5M]
titlebar_items: TVOC_Level
titlebar_name_type: full
calibrate_x_position: 0
calibrate_y_position: 0

And "nano" text editor is installed.

**Comment by DominicWeston** [comment]
What I have done is:
ssh root@10.0.0.14            <--- which is my IP
root@10.0.0.14's password:   <--- which the default is klipper
*._.* Welcome to the ADM5 Klipper Mod *._.*
ADM5 ~ #  ls -laR 

This will list all files and directories. I cut and paste the results into a text file on my Macbook.
That way I can do all the heavy lifting on my Mac and not the printer's cpu.
I delete all the directories and the content for /proc/ directories from the file, this will slim down the text file. 
Then I can search the text file for "timezone"  to find any file names containing timezone.
Then do for the files in the ssh shell:
cat ./mnt/orig_root/usr/share/postgresql/timezonesets/America.txt | grep EST
Gives:
# CONFLICT! EST is not unique
#  - EST: Eastern Standard Time (Australia)
EST    -18000    # Eastern Standard Time (America)

EST "should" work for EST in the U.S. instead of UTC in my first reply.

Modifying the file system which could work or really hose your system.
Do this at your own peril. I know just enough Linux shell to be dangerous.

**Reply by consp** [reply]
The available names are posix standard. You sould select one which is present in the klipperscreen settings menu.

> /mnt/orig_root/usr/share/postgresql/timezonesets/America.txt

Postgresql timezones are not supported and unlikely to work.

From `/usr/share/zoninfo` (which is where the actual timezones are):

```
EST         GB          HST         Japan       NZ          Portugal    UTC
CET         EST5EDT     GB-Eire     Hongkong    Kwajalein   NZ-CHAT     ROC         Universal
CST6CDT     Egypt       GMT         Iceland     Libya       Navajo      ROK         W-SU
Eire        GMT+0       MET         PRC         Singapore   WET
Etc         GMT-0       Iran        MST         PST8PDT     Turkey      Zulu
Cuba        GMT0        Israel      MST7MDT     UCT
EET         Factory     Greenwich   Jamaica     Poland
```

If you want to use the [region]/[capital] variant you have to set it manually. For that you have to manually change the symlink in /root/printer_data/config/timezone to point to the correct file in /usr/share/zoneinfo.

> CONFLICT! EST is not unique

Yes it is. EST is UTC-5 in posix, EST for australia does not exist in posix and would have been named AEST but they decided to ignore that and add australia as a region. Whatever postgres does, does not apply here.

**Comment by DominicWeston** [comment]
There is also a file in /etc/timezone which is more likely the one to modify.

**Reply by consp** [reply]
> There is also a file in /etc/timezone which is more likely the one to modify.

`lrwxrwxrwx    1 root     root            34 Jan  1  1970 /etc/localtime -> /root/printer_data/config/timezone`

/etc/localtime is a symlink to /root/printer_data/config/timezone this is done since the data in /etc is not persistent. You can either change the timezone by symlinking the timezone file in `/usr/share/zoneinfo` or by changing the setting in the klipperscreen config to one in `/usr/share/zoneinfo/posix` (or use the settings menu)

**Comment by consp** [comment]
> The time is incorrect on klipperscreen, but I can't figure out how to change it. Could someone point me in the right direction?

Go to settings and scroll down. There should be a timezone setting.

**Comment by DominicWeston** [comment]
Symlinks! Thanks for the gentle corrections. I'll amend the line "I know just enough Linux shell to be dangerous" to read "I don't know enough Linux shell to be very dangerous".
